File: quick-fix.py
#!/usr/bin/env python3
"""
Script rápido para corrigir erros específicos de TypeScript
"""

import os
import re

def quick_fix(file_path):
    """Correção rápida de erros específicos"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        original = content
        
        # Correções específicas
        content = re.sub(r'(\w+):\s*([^,}]+);\s*\n\s*(\w+):', r'\1: \2,\n                \3:', content)
        content = re.sub(r'(\w+):\s*([^,}]+);\s*\n\s*}', r'\1: \2\n            }', content)
        content = re.sub(r'(\w+):\s*([^,}]+)\s*\n\s*(\w+):', r'\1: \2,\n                \3:', content)
        content = re.sub(r'(\w+):\s*([^,}]+)\s*\n\s*}', r'\1: \2\n            }', content)
        content = re.sub(r'(\w+):\s*([^,)]+);\s*\)', r'\1: \2)', content)
        content = re.sub(r'(\w+):\s*([^,)]+)\s*\)', r'\1: \2)', content)
        content = re.sub(r'const\s+(\w+)\s*=\s*([^;]+);\s*$', r'const \1 = \2;', content, flags=re.MULTILINE)
        content = re.sub(r'const\s+(\w+)\s*=\s*([^,;]+)\s*$', r'const \1 = \2;', content, flags=re.MULTILINE)
        content = re.sub(r'return\s+([^;]+);\s*$', r'return \1;', content, flags=re.MULTILINE)
        content = re.sub(r'return\s+([^,;]+)\s*$', r'return \1;', content, flags=re.MULTILINE)
        content = re.sub(r'if\s*\([^)]+\)\s*{\s*([^}]+);\s*}', r'if (\1) {\n        \2;\n    }', content)
        content = re.sub(r'if\s*\([^)]+\)\s*{\s*([^}]+)\s*}', r'if (\1) {\n        \2;\n    }', content)
        content = re.sub(r'for\s*\([^)]+\)\s*{\s*([^}]+);\s*}', r'for (\1) {\n        \2;\n    }', content)
        content = re.sub(r'for\s*\([^)]+\)\s*{\s*([^}]+)\s*}', r'for (\1) {\n        \2;\n    }', content)
        content = re.sub(r'while\s*\([^)]+\)\s*{\s*([^}]+);\s*}', r'while (\1) {\n        \2;\n    }', content)
        content = re.sub(r'while\s*\([^)]+\)\s*{\s*([^}]+)\s*}', r'while (\1) {\n        \2;\n    }', content)
        content = re.sub(r'case\s+([^:]+):\s*([^;]+);\s*break;', r'case \1:\n            \2;\n            break;', content)
        content = re.sub(r'case\s+([^:]+):\s*([^;]+)\s*break;', r'case \1:\n            \2;\n            break;', content)
        content = re.sub(r'try\s*{\s*([^}]+);\s*}\s*catch', r'try {\n        \1;\n    } catch', content)
        content = re.sub(r'try\s*{\s*([^}]+)\s*}\s*catch', r'try {\n        \1;\n    } catch', content)
        content = re.sub(r'finally\s*{\s*([^}]+);\s*}', r'finally {\n        \1;\n    }', content)
        content = re.sub(r'finally\s*{\s*([^}]+)\s*}', r'finally {\n        \1;\n    }', content)
        content = re.sub(r'async\s+(\w+)\s*\([^)]*\)\s*{\s*([^}]+);\s*}', r'async \1() {\n        \2;\n    }', content)
        content = re.sub(r'async\s+(\w+)\s*\([^)]*\)\s*{\s*([^}]+)\s*}', r'async \1() {\n        \2;\n    }', content)
        content = re.sub(r'(\w+)\s*=\s*\([^)]*\)\s*=>\s*{\s*([^}]+);\s*}', r'\1 = () => {\n        \2;\n    }', content)
        content = re.sub(r'(\w+)\s*=\s*\([^)]*\)\s*=>\s*{\s*([^}]+)\s*}', r'\1 = () => {\n        \2;\n    }', content)
        content = re.sub(r'(\w+)\s*\([^)]*\)\s*{\s*([^}]+);\s*}', r'\1() {\n        \2;\n    }', content)
        content = re.sub(r'(\w+)\s*\([^)]*\)\s*{\s*([^}]+)\s*}', r'\1() {\n        \2;\n    }', content)
        content = re.sub(r'(\w+):\s*([^;]+);\s*\n\s*(\w+):', r'\1: \2;\n    \3:', content)
        content = re.sub(r'(\w+):\s*([^;]+)\s*\n\s*(\w+):', r'\1: \2;\n    \3:', content)
        content = re.sub(r'type\s+(\w+)\s*=\s*([^;]+);\s*$', r'type \1 = \2;', content, flags=re.MULTILINE)
        content = re.sub(r'type\s+(\w+)\s*=\s*([^;]+)\s*$', r'type \1 = \2;', content, flags=re.MULTILINE)
        content = re.sub(r'(\w+)\s*=\s*([^,;]+);\s*\n\s*(\w+)', r'\1 = \2,\n    \3', content)
        content = re.sub(r'(\w+)\s*=\s*([^,;]+)\s*\n\s*(\w+)', r'\1 = \2,\n    \3', content)
        content = re.sub(r'(\w+):\s*([^;]+);\s*\n\s*(\w+):', r'\1: \2;\n    \3:', content)
        content = re.sub(r'(\w+):\s*([^;]+)\s*\n\s*(\w+):', r'\1: \2;\n    \3:', content)
        content = re.sub(r'declare\s+module\s+[\'"][^\'"]*[\'"]\s*{\s*([^}]+);\s*}', r'declare module \'\1\' {\n    \2;\n}', content)
        content = re.sub(r'declare\s+module\s+[\'"][^\'"]*[\'"]\s*{\s*([^}]+)\s*}', r'declare module \'\1\' {\n    \2;\n}', content)
        content = re.sub(r'(\w+)\s*extends\s*([^;]+);\s*$', r'\1 extends \2', content, flags=re.MULTILINE)
        content = re.sub(r'(\w+)\s*extends\s*([^;]+)\s*$', r'\1 extends \2', content, flags=re.MULTILINE)
        content = re.sub(r'(\w+)\s*extends\s*([^?]+)\s*\?\s*([^:]+)\s*:\s*([^;]+);\s*$', r'\1 extends \2 ? \3 : \4', content, flags=re.MULTILINE)
        content = re.sub(r'(\w+)\s*extends\s*([^?]+)\s*\?\s*([^:]+)\s*:\s*([^;]+)\s*$', r'\1 extends \2 ? \3 : \4', content, flags=re.MULTILINE)
        content = re.sub(r'{\s*\[K\s+in\s+keyof\s+(\w+)\]:\s*([^;]+);\s*}', r'{ [K in keyof \1]: \2 }', content)
        content = re.sub(r'{\s*\[K\s+in\s+keyof\s+(\w+)\]:\s*([^;]+)\s*}', r'{ [K in keyof \1]: \2 }', content)
        content = re.sub(r'Partial\s*<\s*(\w+)\s*>;\s*$', r'Partial<\1>', content, flags=re.MULTILINE)
        content = re.sub(r'Required\s*<\s*(\w+)\s*>;\s*$', r'Required<\1>', content, flags=re.MULTILINE)
        content = re.sub(r'Pick\s*<\s*(\w+),\s*(\w+)\s*>;\s*$', r'Pick<\1, \2>', content, flags=re.MULTILINE)
        content = re.sub(r'Omit\s*<\s*(\w+),\s*(\w+)\s*>;\s*$', r'Omit<\1, \2>', content, flags=re.MULTILINE)
        content = re.sub(r'Partial\s*<\s*(\w+)\s*>\s*$', r'Partial<\1>', content, flags=re.MULTILINE)
        content = re.sub(r'Required\s*<\s*(\w+)\s*>\s*$', r'Required<\1>', content, flags=re.MULTILINE)
        content = re.sub(r'Pick\s*<\s*(\w+),\s*(\w+)\s*>\s*$', r'Pick<\1, \2>', content, flags=re.MULTILINE)
        content = re.sub(r'Omit\s*<\s*(\w+),\s*(\w+)\s*>\s*$', r'Omit<\1, \2>', content, flags=re.MULTILINE)
        
        if content != original:
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            return True
        return False
        
    except Exception as e:
        print(f"❌ Erro: {file_path}: {e}")
        return False

def main():
    print("🚀 Correção rápida de erros TypeScript...")
    
    ts_files = []
    for root, dirs, files in os.walk('.'):
        dirs[:] = [d for d in dirs if d not in ['node_modules', '.next', '.vercel']]
        for file in files:
            if file.endswith(('.ts', '.tsx')) and not file.startswith('.'):
                ts_files.append(os.path.join(root, file))
    
    print(f"📊 {len(ts_files)} arquivos encontrados")
    
    fixed = 0
    for file_path in ts_files:
        if quick_fix(file_path):
            print(f"✅ {file_path}")
            fixed += 1
        else:
            print(f"⏭️  {file_path}")
    
    print(f"\n✅ Concluído! {fixed} arquivos corrigidos de {len(ts_files)}")

if __name__ == "__main__":
    main()

File: quick-fix.js
#!/usr/bin/env node

/**
 * 🚀 Quick Fix - Correção Rápida de Erros de Build
 * Script simples e direto para corrigir os erros mais comuns
 */

const fs = require('fs');
const path = require('path');
const { execSync } = require('child_process');

console.log('🚀 Quick Fix - Corrigindo erros de build...\n');

// Lista de correções específicas baseadas nos erros que vimos
const quickFixes = [
    {
        name: 'Monaco Editor Types',
        files: ['frontend/components/AdvancedEditorSimple.tsx'],
        fixes: [
            {
                from: /lineNumbers:\s*(\w+)\s*\?\s*['"`]on['"`]\s*:\s*['"`]off['"`]/g,
                to: "lineNumbers: $1 ? 'on' as const : 'off' as const"
            },
            {
                from: /wordWrap:\s*(\w+)\s*\?\s*['"`]on['"`]\s*:\s*['"`]off['"`]/g,
                to: "wordWrap: $1 ? 'on' as const : 'off' as const"
            }
        ]
    },
    {
        name: 'ProfileStorage Methods',
        files: ['frontend/app/profile/useUserProfile.ts'],
        fixes: [
            {
                from: /ProfileStorage\.saveProfile\(/g,
                to: 'ProfileStorage.save('
            },
            {
                from: /ProfileStorage\.getProfile\(/g,
                to: 'ProfileStorage.load('
            },
            {
                from: /ProfileStorage\.updateProfile\(/g,
                to: 'ProfileStorage.save('
            }
        ]
    },
    {
        name: 'UserProfile Interface',
        files: ['frontend/lib/profileStorage.ts'],
        fixes: [
            {
                from: /(\s+email: string;\s+)/,
                to: '$1  phone?: string;\n'
            },
            {
                from: /(\s+notifications: boolean;\s+)/,
                to: '$1    timezone?: string;\n    emailNotifications?: boolean;\n    pushNotifications?: boolean;\n'
            }
        ]
    },
    {
        name: 'Permissions Types',
        files: ['frontend/app/utils/permissions.ts'],
        fixes: [
            {
                from: /newLevel\s*!==\s*['"`]free['"`]/g,
                to: 'true'
            }
        ]
    },
    {
        name: 'CourseItem Interface',
        files: ['frontend/lib/payment-service.ts'],
        fixes: [
            {
                from: /(export interface PaymentIntent[^}]+}\s*)/,
                to: `$1
export interface CourseItem {
    id: string;
    title: string;
    description: string;
    price: number;
    originalPrice?: number;
    discount?: number;
    currency: string;
    image?: string;
    features?: string[];
    duration?: string;
    level?: string;
    category?: string;
}
`
            }
        ]
    }
];

// Função para aplicar correções
function applyFixes() {
    let totalFixed = 0;

    for (const fixGroup of quickFixes) {
        console.log(`📝 Aplicando: ${fixGroup.name}`);

        for (const filePath of fixGroup.files) {
            if (fs.existsSync(filePath)) {
                let content = fs.readFileSync(filePath, 'utf8');
                let hasChanges = false;

                for (const fix of fixGroup.fixes) {
                    const before = content;
                    content = content.replace(fix.from, fix.to);
                    if (before !== content) {
                        hasChanges = true;
                    }
                }

                if (hasChanges) {
                    fs.writeFileSync(filePath, content, 'utf8');
                    console.log(`  ✅ Corrigido: ${filePath}`);
                    totalFixed++;
                }
            } else {
                console.log(`  ⚠️  Arquivo não encontrado: ${filePath}`);
            }
        }
    }

    return totalFixed;
}

// Função para executar build
function runBuild() {
    try {
        console.log('\n🔨 Executando build...');
        execSync('npm run build', {
            cwd: 'frontend',
            stdio: 'inherit'
        });
        console.log('\n✅ Build executado com sucesso!');
        return true;
    } catch (error) {
        console.log('\n❌ Build falhou. Erros ainda persistem.');
        return false;
    }
}

// Função para executar deploy
function runDeploy() {
    try {
        console.log('\n🚀 Executando deploy...');
        execSync('vercel --prod --yes', {
            cwd: 'frontend',
            stdio: 'inherit'
        });
        console.log('\n✅ Deploy executado com sucesso!');
        return true;
    } catch (error) {
        console.log('\n❌ Deploy falhou.');
        return false;
    }
}

// Executar correções
const fixedCount = applyFixes();
console.log(`\n📊 Total de arquivos corrigidos: ${fixedCount}`);

// Executar build
const buildSuccess = runBuild();

if (buildSuccess) {
    // Perguntar sobre deploy
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question('\n🚀 Deseja fazer deploy para produção? (s/n): ', (answer) => {
        if (answer.toLowerCase() === 's' || answer.toLowerCase() === 'sim') {
            runDeploy();
        } else {
            console.log('\n✅ Correções aplicadas com sucesso!');
            console.log('💡 Para fazer deploy manualmente: vercel --prod --yes');
        }
        rl.close();
    });
} else {
    console.log('\n⚠️  Execute este script novamente para corrigir erros restantes.');
}














